catalog/plugin/catalog-plugin-metacardingest-network/src/test/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeHelperTest.java

```diff
@@ -0,0 +1,351 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.when;
+import static ddf.catalog.data.AttributeType.AttributeFormat.BINARY;
+import static ddf.catalog.data.AttributeType.AttributeFormat.BOOLEAN;
+import static ddf.catalog.data.AttributeType.AttributeFormat.DATE;
+import static ddf.catalog.data.AttributeType.AttributeFormat.DOUBLE;
+import static ddf.catalog.data.AttributeType.AttributeFormat.FLOAT;
+import static ddf.catalog.data.AttributeType.AttributeFormat.GEOMETRY;
+import static ddf.catalog.data.AttributeType.AttributeFormat.INTEGER;
+import static ddf.catalog.data.AttributeType.AttributeFormat.LONG;
+import static ddf.catalog.data.AttributeType.AttributeFormat.OBJECT;
+import static ddf.catalog.data.AttributeType.AttributeFormat.SHORT;
+import static ddf.catalog.data.AttributeType.AttributeFormat.STRING;
+import static ddf.catalog.data.AttributeType.AttributeFormat.XML;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+
+/**
+ * Validate the attribute methods that modify metacards.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class AttributeHelperTest {
+
+    private static final String SAMPLE_JSON_STRING = "{ \"id\": \"j8k767bjav592j2\"}";
+
+    private static final String METACARD_TITLE = "Metacard Title";
+
+    private static final String METACARD_DESCRIPTION = "Hello from the metacard";
+
+    private static final String DESCRIPTION_KEY = "description";
+
+    @Mock
+    private AttributeDescriptor mockDescriptor;
+
+    @Mock
+    private AttributeType mockType;
+
+    private AttributeHelper attributeHelper;
+
+    @Before
+    public void setup() throws Exception {
+        when(mockDescriptor.getType()).thenReturn(mockType);
+        attributeHelper = new AttributeHelper();
+    }
+
+    @Test
+    public void testApplyNewAttributesOnEmptyList() throws Exception {
+        List<Metacard> metacards = ImmutableList.of();
+        Map<String, String> attributeMap = ImmutableMap.of();
+
+        attributeHelper.applyNewAttributes(metacards, attributeMap);
+    }
+
+    @Test
+    public void testApplyNewAttributesWithEmptyMap() throws Exception {
+        Metacard metacard = createMetacard();
+        List<Metacard> metacards = ImmutableList.of(metacard);
+        Map<String, String> attributeMap = ImmutableMap.of();
+
+        attributeHelper.applyNewAttributes(metacards, attributeMap);
```

Comment:
Should we verify that we have the expected number of attributes in the `metacard` before doing the other verifications (same question for all the other tests)?